builtins.go

```diff
@@ -1425,6 +1426,143 @@ func builtinParseYAML(i *interpreter, str value) (value, error) {
 	return jsonToValue(i, elems[0])
 }
 
+func builtinParseCSV(i *interpreter, str value) (value, error) {
+	sval, err := i.getString(str)
+	if err != nil {
+		return nil, err
+	}
+	s := sval.getGoString()
+
+	json := make([]interface{}, 0)
+	var keys []string
+
+	reader := csv.NewReader(strings.NewReader(s))
+	for row := 0; ; row++ {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, i.Error(fmt.Sprintf("failed to parse JSON: %s", err.Error()))
+		}
+
+		if row == 0 { // consider first row as header
+			keys = record
+		} else {
+			j := make(map[string]interface{})
+			for i, k := range keys {
+				j[k] = record[i]
```

Comment:
This seems a bit opinionated.  Is there a precedent for this way of doing it?  E.g. what if the column header is the empty string or just that 2 columns have a duplicate column header.  Seems like this would only let you parse a CSV it it followed a pattern of having unique column headers.

Reply:
If you want that semantics it'd be better to call it `parseCsvWithHeader` or something like that.  Then we could add a more generic one that just returns an array of arrays as `parseCsv` later.

Reply:
another option: `parseCsvToObjects`